src/validation.cpp

```diff
@@ -4824,6 +4824,28 @@ std::vector<Chainstate*> ChainstateManager::GetAll()
     return out;
 }
 
+std::vector<Chainstate*> ChainstateManager::GetAllForBlockDownload()
+{
+    AssertLockHeld(::cs_main);
+    std::vector<Chainstate*> out;
+
+    bool snapshot_in_ibd{
+        m_snapshot_chainstate && m_snapshot_chainstate->IsInitialBlockDownload()};
+
+    if (m_snapshot_chainstate) {
+        out.push_back(m_snapshot_chainstate.get());
+    }
+    // Exclude the ibd chainstate (which is in the background if we have a snapshot
+    // chainstate active) if the snapshot chainstate is in IBD because the main
+    // priority is getting the active chain to network tip.
```

Comment:
In commit "validation: add ChainstateManager::GetAllForBlockDownload()" (42f5b6b03bba1ef1c1bc6c225606d119c4e27169)

Is there a reason to exclude the ibd chainstate if the comment in commit "net_processing: work with multiple chainstates" (c5bfecff4d0db726fab893a46cc1dab29591f325) is correct: "The first chainstate in line for processing will likely exhaust this during IBD, but once it hits a tip capacity will trickle into subsequent chainstates."?

It seems cleaner if code in net_processing just decide what blocks to download and validation code doesn't have to be involved in the details.